### ISSUE TITLE
[Roadmap] GeRL Roadmap Q1 2026

### ISSUE THREAD

**Issue by chenyingshu** [leibniz-csi/mm_grpo#48]
# GeRL

**GeRL** (multi-modal **Ge**nerative model **RL** framework) is an easy-to-use and fast library to support Reinforcement Learning (RL) training for multi-modal generative models. It is built on top of [verl](https://github.com/volcengine/verl), [vLLM-Omni](https://github.com/vllm-project/vllm-omni) and [diffusers](https://github.com/huggingface/diffusers), aiming to streamline RL fine-tuning for advanced multi-modal generative models.

---

## 🎯 Project Overview

- **Goal**: Provide a scalable and user-friendly framework for RL training of multi-modal generative models.
- **Core Features**:
  - Support RL training for generative models such as T2I diffusion models.
  - Compatible with diffusion models in `diffusers`.
  - Compatible with multi-modal models in `vllm-omni`.
  - Support diverse RL algorithms for generative models.
  - Efficient parallel training with asynchronous scheduling, and ED disaggregation..

---

## 🗓️ Roadmap

This is the second roadmap for GeRL v0.1.0.

#### Model Support

- [ ] DiffusionNFT (SD3.5)
- [ ] Compatible with diffusers-based pipelines.
- [ ] [vLLM-Omni](https://github.com/vllm-project/vllm-omni) models support for rollout (QwenImage, etc.)

#### Reinforcement Learning Algorithms

- [ ] DiffusionNFT 🔥

#### Training

- [ ] Full support for DiffusionNFT training pipeline, including performance evaluation.
- [ ] Implement and test reward models for DiffusionNFT.

#### RL Efficiency and Improvements🔥

- [ ] **Fully asynchronous scheduling** for RL training generative models.

#### Documentation & Examples

- [ ] Provide tutorials and scripts for DiffusionNFT fine-tuning.
- [ ] Development and contribution guidelines for developers

---

#### Model Support

- [ ] QwenImage
- [ ] HunyuanImage-3.0
- [ ] Compatible with more diffusers-based pipelines

#### Reinforcement Learning Algorithms

- [ ] [DGPO](https://github.com/Luo-Yihong/DGPO)
- [ ] MixGRPO (HunyuanImage-3.0)
- [ ] SRPO (HunyuanImage-3.0)

#### RL Efficiency and Improvements🔥

- [ ] Support reward model with separate resource pool.
- [ ] Encoder-Decoder disaggregation to improve training efficiency.
- [ ] Optimize hyperparameters for stable training.

#### Community and Ecosystem

- [ ] Encourage community contributions for model and algorithm support.
- [ ] Build baselines for different RL algorithms and training strategies.

---

## 🤝 How to Contribute

We welcome contributions from the community! Feel free to:

- Report issues or suggest features.
- Submit pull requests for bug fixes or new features.
- Share your use cases.

For more details, please refer to [CONTRIBUTING.md](https://github.com/leibniz-csi/mm_grpo/blob/main/CONTRIBUTING.md).

## 📬 Contact

For questions or collaborations, please open an issue or contact the maintainers.
